build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -0,0 +1,267 @@
+                Constants.OryxSdkStorageBaseUrl + DotNetConstants.DotNetSuffixUrl);
+            this.dotNetMinReleaseVersion = Environment.GetEnvironmentVariable(DotNetConstants.DotNetMinReleaseVersionEnvVar);
+            this.dotNetMaxReleaseVersion = Environment.GetEnvironmentVariable(DotNetConstants.DotNetMaxReleaseVersionEnvVar);
+            var blockedVersions = Environment.GetEnvironmentVariable(
+                DotNetConstants.DotNetBlockedVersionsEnvVar);
+            if (!string.IsNullOrEmpty(blockedVersions))
+            {
+                var versionStrings = blockedVersions.Split(',');
+                foreach (var versionString in versionStrings)
+                {
+                    this.dotNetBlockedVersions.Add(versionString.Trim());
+                }
+            }
+
+            List<DotNetVersion> newDotNetVersions = await this.GetNewDotNetVersionsAsync();
+            if (newDotNetVersions.Count > 0)
+            {
+                string constantsYamlSubPath = Path.Combine("build", Constants.ConstantsYaml);
+                List<ConstantsYamlFile> yamlConstantsObjs = await this.yamlFileService.ReadConstantsYamlFileAsync(constantsYamlSubPath);
+                this.UpdateOryxConstantsForNewVersions(newDotNetVersions, yamlConstantsObjs);
+            }
+        }
+
+        public async Task<List<DotNetVersion>> GetNewDotNetVersionsAsync()
+        {
+            List<DotNetVersion> dotNetVersions = new List<DotNetVersion>();
+
+            // Deserialize release metadata
+            var response = await this.httpService.GetDataAsync(DotNetConstants.ReleasesIndexJsonUrl);
+            var releaseNotes = response == null ? null : JsonConvert.DeserializeObject<ReleaseNotes>(response);
+            var releasesIndex = releaseNotes == null ? new List<ReleaseNote>() : releaseNotes.ReleaseIndexes;
+            foreach (var releaseIndex in releasesIndex)
+            {
+                // If the latest release version is not within the acceptable range (inclusive),
+                // or if the sdk version is present in the set of ORYX SDK versions,
+                // skip to the next version in the loop.
+                if (this.ReleaseVersionIsNotInRangeOrSdkVersionAlreadyExists(
+                    releaseIndex.LatestRelease, releaseIndex.LatestSdk))
+                {
+                    continue;
+                }
+
+                // Get the actual release information from releases.json
+                string releasesJsonUrl = releaseIndex.ReleasesJsonUrl;
+                response = await this.httpService.GetDataAsync(releasesJsonUrl);
+                var releasesJson = JsonConvert.DeserializeObject<ReleasesJson>(response);
+                var releases = releasesJson == null ? new List<Release>() : releasesJson.Releases;
+                foreach (var release in releases)
+                {
+                    // Check again since the "release" is separate from "releaseIndex".
```

Comment:
This would only happen if the .NET team had messed up versions (something like the .NET 7 release index containing .NET 6 releases), right? If it's the case, then please consider throwing an exception as this should never happen, and if it does we should let them know.

Reply:
We need to check our storage account twice because on how .NET releases information is structured. Both release index and releases can have versions that we should check for in our storage account. 

Here is an example flow:
1. GET https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json
a. Check storage account once to make sure the latest version is not already in storage account
b. This is the release index
3. Get releases URL from `"releases.json"`
4. GET https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/7.0/releases.json
a. Check storage account again. Since since the `"releases"` elements has all `7.0` versions

Reply:
OK fair enough. I was mostly focusing on validating that the version is correct, but it's true that we also need to check if things already exist.